core/src/main/scala/org/apache/spark/executor/MesosExecutorBackend.scala

```diff
@@ -81,6 +90,12 @@ private[spark] class MesosExecutorBackend
       env)
   }
 
+  private def getRangesFromResources(executorInfo:ExecutorInfo): List[(Long, Long)] ={
+  executorInfo.getResourcesList.asScala.filter(r => r.getType == Value.Type.RANGES & r.getName == "ports").toList.
+      map(r => r.getRanges.getRangeList).flatMap { ranges => ranges.asScala.map { r => (r.getBegin, r.getEnd) }}
+  }
```

Comment:
Nit picking: Do you have to check for Type.RANGES, can't we just check the name "ports"? Using asScala.find might be better. As it is consistent with the other parts of the code.